---
title: "Beas-2b SLC25A4 KD GSEA"
author: "Libby Murphy"
date: "2023-10-05"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
library(tidyverse)
library(BiocManager)
library(DESeq2)
library(dplyr)
library(fgsea)
```

```{r build geneInfo}
geneInfo <- read_csv("/Users/libbymurphy/Documents/GitHub/Konigshoff_ANT1KD_RNAseq/accessories/gencode.v26.primary.info.csv.zip", col_names = F)

colnames(geneInfo) <- c("gene_id", "transcript_id", "biotype", "symbol")

geneInfo$gene_id <- sapply(
  strsplit(
    geneInfo$gene_id,".", fixed = T
  ),
  function(x) x[1]
)

geneInfo <- geneInfo[,c(1,3,4)] %>% unique()
```

```{r read in file and build DESeq object}
antfarmking <- read_tsv("/Users/libbymurphy/Documents/GitHub/Konigshoff_ANT1KD_RNAseq/data/GSE244305_Ant1ko_bleo_study_gene_count.txt") %>%
  as.data.frame()

rownames(antfarmking) <- antfarmking[,1]

coldata <- data.frame(matrix(nrow = length(colnames(antfarmking)[2:13]),
                             ncol = 2)
                      )
coldata <- data.frame(
  "sample" = colnames(antfarmking[2:13])
) %>%
  separate(
    col = sample, 
    into = c("condition", "tx", "rep")
  )

coldata$tx <- coldata$tx %>%
  gsub(
    pattern = "Bleo",
    replacement = "bleo"
  )

rownames(coldata) <- colnames(antfarmking)[2:13]

coldata$condition <- factor(coldata$condition, levels = c("WT", "A1KO"))
coldata$tx <- factor(coldata$tx, levels = c("ctrl", "bleo"))
coldata$rep <- factor(coldata$rep, levels = c("1", "2", "3"))

stopifnot(
  all(colnames(antfarmking[,2:13]) %in% rownames(coldata)) == TRUE & 
  all(colnames(antfarmking[,2:13]) == rownames(coldata)) == TRUE
)



dds_ANT1KD <- DESeqDataSetFromMatrix(
  countData = antfarmking[,2:13],
  colData = coldata,
  design = ~ condition
)

dds_ANT1KD$condition <- factor(dds_ANT1KD$condition, levels = c("WT", "A1KO"))

keep <- rowSums(counts(dds_ANT1KD)) >= 10
dds_ANT1KD <- dds_ANT1KD[keep,]

dds_ANT1KD <- DESeq(dds_ANT1KD)

resultsNames(dds_ANT1KD)

res_bleo <- results(dds_ANT1KD, contrast = c("condition", "A1KO", "WT"))

res_bleo <- res_bleo %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_id")

lfcInfo <- dplyr::left_join(
  res_bleo,
  geneInfo,
  by = "gene_id"
)[,c(9,1:8)]
```

```{r filter genes}
biotypes <- unique(lfcInfo$biotype)[c(1,3,17,19,21)]

lfcInfo_filtered <- lfcInfo %>%
  dplyr::filter(biotype %in% biotypes)

lfcInfo_filtered <- lfcInfo_filtered %>%
  dplyr::filter(
    log2FoldChange <= 0.5
  )

write_csv(
  lfcInfo_filtered,
  path = "/Users/libbymurphy/Documents/GitHub/Konigshoff_ANT1KD_RNAseq/results/lfc_filtered.csv"
)
```

This output table represents log2 fold change of ANT1 KO cells as compared to WT, filtered for genomic annotation and p-value <= 0.5. 

```{r GSEA}

```
