---
title: "Beas-2b SLC25A4 KO GSEA"
author: "Libby Murphy"
date: `r Sys.Date()`
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
library(tidyverse)
library(here)
library(readr)
library(BiocManager)
library(DESeq2)
library(dplyr)
library(clusterProfiler)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(ggplot2)
library(GenomicFeatures)
library(biomaRt)
```

```{r build geneInfo and counts_data}
counts_data <- read_tsv(here("data", "GSE244305_Ant1ko_bleo_study_gene_count.txt")) %>%
  as.data.frame()

colnames(counts_data) <- colnames(counts_data) %>%
  gsub(pattern = "Bleo", replacement = "bleo")

# saveRDS(counts_data, file = here("accessories", "counts_data.RDS"))
# sometimes read_tsv doesn't work for an unknown reason--read in rds instead if needed

rownames(counts_data) <- counts_data[,1]

mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))

genes <- counts_data$gene_id

geneInfo <- getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id","hgnc_symbol"),values=genes,mart= mart)

colnames(geneInfo) <- c("gene_id", "symbol")
```

```{r PCA plot QC}
qcinput <- counts_data[,(2:13)] %>% 
  as.matrix()

qcinput <- log2(qcinput[rowSums(qcinput) > 20,] + 1)

pca_data <- prcomp(qcinput, center = T, scale. = T) 

pca_data_info <- summary(pca_data) 

pca_plot_data <- data.frame(pca_data$rotation) 

pca_plot_data$ID <- rownames(pca_plot_data)

pca_plot_data <- pca_plot_data %>% 
  separate(col = ID, sep = "_", into = c("genotype","bleo","rep"))

labelPosition <- pca_plot_data %>% 
  group_by(genotype, bleo, rep) %>% 
  dplyr::select(PC1, PC2) %>% 
  summarise(mPC1=mean(PC1), mPC2=mean(PC2))

myCols <- c("darkgrey",
            RColorBrewer::brewer.pal(n = 8, name = "Blues")[6],
            RColorBrewer::brewer.pal(n = 6, name = "Reds")[c(3,6)])


PCAplot <- ggplot(pca_plot_data, aes(x=PC1, y=PC2, color = bleo, shape = genotype)) +
  geom_point(size=3) + 
  theme_classic() +
  ggtitle("PCA of Konigshoff Beas-2b ANT1 KO") +
  scale_color_manual(values = myCols) +
  xlab(paste("PC1 (%",100*round(pca_data_info$importance[2,1], digits = 3),")", sep = "")) +
  ylab(paste("PC2 (%",100*round(pca_data_info$importance[2,2], digits = 3),")", sep = "")) 

PCAplot

# ggplot2::ggsave(filename = here("results", "PCAplot_QC.png"), plot = PCAplot, device = "png")
```

```{r build DESeq object and metadata}
coldata <- data.frame(matrix(nrow = length(colnames(counts_data)[2:13]),
                             ncol = 2)
                      )

coldata <- data.frame(
  "sample" = colnames(counts_data[2:13])
) %>%
  separate(
    col = sample, 
    into = c("condition", "tx", "rep")
  )


rownames(coldata) <- colnames(counts_data)[2:13]

coldata$condition <- factor(coldata$condition, levels = c("WT", "A1KO"))
coldata$tx <- factor(coldata$tx, levels = c("ctrl", "bleo"))
coldata$rep <- factor(coldata$rep, levels = c("1", "2", "3"))

stopifnot(
  all(colnames(counts_data[,2:13]) %in% rownames(coldata)) == TRUE & 
  all(colnames(counts_data[,2:13]) == rownames(coldata)) == TRUE
)
```

``` {r deseq and results}
dds_ANT1KD <- DESeqDataSetFromMatrix(
  countData = counts_data[,2:13],
  colData = coldata,
  design = ~ condition
)

dds_ANT1KD$condition <- factor(dds_ANT1KD$condition, levels = c("WT", "A1KO"))

keep <- rowSums(counts(dds_ANT1KD)) >= 10

dds_ANT1KD <- dds_ANT1KD[keep,]

dds_ANT1KD <- DESeq(dds_ANT1KD)

resultsNames(dds_ANT1KD)

res_bleo <- results(dds_ANT1KD, contrast = c("condition", "A1KO", "WT"))
```

```{r GO}
# using only significantly enriched genes
tmp <- res_bleo[!is.na(res_bleo$padj),]
tmp <- tmp[(tmp$padj <= 0.05),]
tmp <- tmp[(tmp$log2FoldChange >= 0.5),]
tmp <- tmp %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_id")
left_join(tmp, geneInfo, by = "gene_id")

enriched_genes <- rownames(res_bleo[res_bleo$log2FoldChange >= 0.5,])

GO_results <- as.data.frame(enrichGO(
  gene = enriched_genes,
  OrgDb = "org.Hs.eg.db",
  keyType = "ENSEMBL",
  ont = "BP"
))

# view(GO_results)
```

```{r GSEA}
pre_gsea <- res_bleo[res_bleo$baseMean > 50,]
pre_gsea <- pre_gsea[order(-pre_gsea$log2FoldChange),]

gene_list <- pre_gsea$log2FoldChange
names(gene_list) <- rownames(pre_gsea)

gsea_BP <- gseGO(
  geneList = gene_list,
  ont = "BP",
  keyType = "ENSEMBL",
  OrgDb = "org.Hs.eg.db",
  eps = 1e-300
)

gseaplot(
  gsea_BP, 
  geneSetID = 13
)
```
